baseapp/server.go

```diff
@@ -102,6 +102,13 @@ func (s *Server) Start() error {
 
 	addr := s.config.Address + ":" + strconv.Itoa(s.config.Port)
 	s.logger.Info().Msgf("Server listening on %s", addr)
+
+	tlsConfig := s.config.TLSConfig
+
+	if tlsConfig != nil {
+		return http.ListenAndServeTLS(addr, s.config.TLSConfig.CertificateFile, s.config.TLSConfig.PrivateKeyFile, s.mux)
```

Comment:
does validation (existing and being valid) take place on first request or at this call? if on first request, shoudl we at least check file exists?

Reply:
`ListenAndServeTLS` does a check in for the existence and validity of the files, returning an `error` when they are invalid. Depending on conventions, I'm happy to add validation here too

Reply:
naw, just wanted to make sure it wouldnt wait to fail on first request. lgtm

Reply:
one last bit: why use `s.config.TLSConfig.CertificateFile` instead of just `tlsConfig.CertificateFile`

Reply:
Whoops, my bad. I meant to update that